### DISCUSSION TITLE
LED Indication for Presets

### DISCUSSION THREAD

**Discussion by Anaximander78** [shanteacontrols/OpenDeck#118] [Q&A]
Hey, i am trying to get 5 buttons to activate 5  presets. LEDs just work with normal "Program change" in Button Configurator. When i change to "Change Preset", LEDs will not work. 

In Control Type  Dropdown menu of LED there is only "Program change (single value)" But no "Local - Program change". If they are the same i dont understand why it is not working. 
I have set the activation id correctly as far i know, otherwise configured Button as "Program change" would not activate LED.

Any idea what i am doing wrong? 

Greetings from Berlin,

Simon

**Comment by Anaximander78** [comment]
The Presets are changed. But Led does not light up

**Reply by paradajz** [reply]
Have you configured the LEDs to react on Program Change in each preset you want to indicate?

**Reply by paradajz** [reply]
Nevermind. The feature is broken. I'll fix it soon.

**Reply by Anaximander78** [reply]
Thanks for getting back at me so soon! Everything else works for me.

**Comment by paradajz** [comment]
New release is now out:
https://github.com/shanteacontrols/OpenDeck/releases/tag/v6.0.5

**Reply by Anaximander78** [reply]
Just a small report back with new update. In my configuration i have set up Encoder32 to change preset in all 5 Presets. The same for LED 8-12 set to program change. I have set all activation IDs of the LEDs to 0. Otherwise it wont work. Also a little erratic is that LED Testing is activated on preset change  on "Red.  Thanks

**Reply by Anaximander78** [reply]
Sorry i mean Encoder31

**Comment by paradajz** [comment]
There's now a new release with some more LED control tweaks:
https://github.com/shanteacontrols/OpenDeck/releases/tag/v6.1.0

**Reply by Anaximander78** [reply]
Jesus, how fast are you?? Amazing

**Comment by Anaximander78** [comment]
Hey, just a heads up. Preset indication stopped working again in 6.2.2, so i went back to 6.1.0

**Reply by paradajz** [reply]
Can't really debug in more detail right now, but this seemed to be an issue with configurator. Try reloading it and setting the correct control modes for LEDs (program change and preset change have been reversed). I've updated the configurator just now.

**Comment by Anaximander78** [comment]
Just for reference:

LED indication stopped working correctly after 6.1.0. If you use this version you are bound to use only 5  presets, not 10.

**Reply by paradajz** [reply]
Can you please send me a video showing the issue with the latest firmware (v6.4.0)?

**Reply by Anaximander78** [reply]
Cant male a viedeo right now, but basically its that i set one encoder to change presets in all presets and LEDs 1-10 to react on preset change in all presets. This configuration worked fine in 6.1.0. Thats it

**Reply by paradajz** [reply]
Should be fixed now, although I won't release the firmware just yet as I want to finish more things first. Which board are you using? I'll attach the firmware here for you to try it out.

**Reply by Anaximander78** [reply]
Thanks a lot! I am using official opendeck v.2 
I am also happy using the 6.1.0 for now, so no worries

**Reply by paradajz** [reply]
[opendeck2.zip](https://github.com/shanteacontrols/OpenDeck/files/8188692/opendeck2.zip)

Test version. All the tests are passing but there are many internal changes in this firmware so beware.

**Reply by Anaximander78** [reply]
Hello, could test the file you have sent me tonight. Does not work. Perhaps i gave you the wrong information? My opendeck board has the version number 2.0.1 not 2.0.0. maybe that is why?  Returned to 6.1.0 and it works. Thanks for your effort again!

**Comment by paradajz** [comment]
Sorry but I need more information. What exactly does not work? Did you set the MIDI channel to 17?

**Comment by Anaximander78** [comment]
I was not aware there are more than 16 channels. Do you mean to set the encoder which sends preset change message to set to

**Reply by Anaximander78** [reply]
a certain channel?

**Reply by paradajz** [reply]
Sorry, I thought I was responding to other thread. Brainfart. Very weird. I'll get back to this tomorrow.

**Comment by paradajz** [comment]
Here. This was a quite devious bug and something I haven't covered in my tests.

[opendeck2.zip](https://github.com/shanteacontrols/OpenDeck/files/8257801/opendeck2.zip)

**Reply by Anaximander78** [reply]
Works really nice. Thanks a lot, again!

**Comment by paradajz** [comment]
Fix part of a new release: https://github.com/shanteacontrols/OpenDeck/releases/tag/v7.0.0
